src/Tests/Microsoft.DotNet.PackageInstall.Tests/NuGetPackageInstallerTests.cs

```diff
@@ -0,0 +1,204 @@
+
+            await Assert.ThrowsAsync<NuGetPackageInstallerException>(() =>
+                _installer.DownloadPackageAsync(
+                    TestPackageId,
+                    new NuGetVersion(TestPackageVersion),
+                    new PackageSourceLocation(overrideSourceFeeds: new[] {nonExistFeed.Value})));
+        }
+
+        [Fact]
+        public async Task GivenNugetConfigInstallSucceeds()
+        {
+            FilePath nugetConfigPath = GenerateRandomNugetConfigFilePath();
+            FileSystemWrapper fileSystem = new FileSystemWrapper();
+            WriteNugetConfigFileToPointToTheFeed(fileSystem, nugetConfigPath);
+
+            string packagePath = await _installer.DownloadPackageAsync(
+                TestPackageId,
+                new NuGetVersion(TestPackageVersion),
+                new PackageSourceLocation(nugetConfigPath));
+            File.Exists(packagePath).Should().BeTrue();
+        }
+
+        [Fact]
+        public async Task GivenAValidNugetConfigAndFailedSourceItShouldError()
+        {
+            DirectoryPath nonExistFeed =
+                new DirectoryPath(Path.GetTempPath()).WithSubDirectories(Path.GetRandomFileName());
+
+            FilePath validNugetConfigPath = GenerateRandomNugetConfigFilePath();
+            FileSystemWrapper fileSystem = new FileSystemWrapper();
+            WriteNugetConfigFileToPointToTheFeed(fileSystem, validNugetConfigPath);
+
+            // "source" option will override everything like nuget.config just like "dotner restore --source ..."
+            await Assert.ThrowsAsync<NuGetPackageInstallerException>(() =>
+                _installer.DownloadPackageAsync(
+                    TestPackageId,
+                    new NuGetVersion(TestPackageVersion),
+                    new PackageSourceLocation(validNugetConfigPath,
+                        overrideSourceFeeds: new[] {nonExistFeed.Value})));
+        }
+
+        [Fact]
+        public async Task GivenAConfigFileRootDirectoryPackageInstallSucceedsViaFindingNugetConfigInParentDir()
+        {
+            FilePath nugetConfigPath = GenerateRandomNugetConfigFilePath();
+            DirectoryPath directoryBelowNugetConfig = nugetConfigPath.GetDirectoryPath().WithSubDirectories("subDir");
+            Directory.CreateDirectory(directoryBelowNugetConfig.Value);
+
+            FileSystemWrapper fileSystem = new FileSystemWrapper();
+            WriteNugetConfigFileToPointToTheFeed(fileSystem, nugetConfigPath);
+
+            string packagePath = await _installer.DownloadPackageAsync(
+                TestPackageId,
+                new NuGetVersion(TestPackageVersion),
+                new PackageSourceLocation(rootConfigDirectory: directoryBelowNugetConfig));
+            File.Exists(packagePath).Should().BeTrue();
+        }
+
+        [Fact]
+        public async Task GivenAllButNoPackageVersionItCanInstallThePackage()
```

Comment:
This test name confuses me, maybe `GivenNoPackageVersionItCanInstallLatestVersionOfPackage`? Also, can we assert that the expected version gets resolved here?